### DISCUSSION TITLE
Is it possible function in variable

### DISCUSSION THREAD

**Discussion by lekpamartin** [Dynatrace/dynatrace-configuration-as-code#1150] [General]
Hi,
How to declare a function with monaco.

I want to use function like "lower" to alter variable. I have a variable with upercase and space and want to use it as id. I have to define 2 more variables

"name": "{{ .global.project }} SLO for {{ .global.service_name }}",
"metricName": "{{ .global.project_lower }} SLO for {{ .global.service_name_transform }}",

For example with jinja it will be 
"name": "{{ .global.project }} SLO for {{ .global.service_name }}",
"metricName": "{{ .global.project | lower }} SLO for {{ .global.service_name | replace(" ","_") | lower }}",

I get undefined function lower when deploy config in monaco 2.

Thx

**Comment by warber** [comment]
Hi,
For that, I'd recommend getting familiar with GO templating.
You can read about what's possible in the "advanced use cases" section in the configuration as code documentation:

https://www.dynatrace.com/support/help/manage/configuration-as-code/guides/configuration-as-code-advanced-use-case

Best
